src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/DoNotUseStackallocInLoops.cs

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Runtime
+{
+    /// <summary>CA2014: Do Not Use Stackalloc In Loops.</summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class DoNotUseStackallocInLoopsAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string RuleId = "CA2014";
+
+        internal static readonly DiagnosticDescriptor Rule = DiagnosticDescriptorHelper.Create(
+            RuleId,
+            new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.DoNotUseStackallocInLoopsTitle), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources)),
+            new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.DoNotUseStackallocInLoopsMessage), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources)),
+            DiagnosticCategory.Reliability,
+            RuleLevel.BuildWarning,
+            new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.DoNotUseStackallocInLoopsDescription), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources)),
+            isPortedFxCopRule: false,
+            isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        private static readonly ImmutableArray<SyntaxKind> s_stackallocKinds = ImmutableArray.Create(SyntaxKind.StackAllocArrayCreationExpression);
+
+        public sealed override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.RegisterCompilationStartAction(compilationContext =>
```

Comment:
Out of curiosity, I have seen this pattern quite often but I am not sure to understand what the point when there is no access to some user-option or known type, could you let me know if there is a special reason?

Reply:
I'm not understanding the question. What pattern are you referring to? What alternative are you expecting?

Reply:
My bad, I meant using the `RegisterCompilationStartAction` before the `RegisterSyntaxNodeAction` without doing anything in between while you could register the syntax node action directly.

Reply:
Ah, no reason, other than I started with as a template another rule that did this.  I can change it.